run.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
```

Comment:
Since you are already using golang, maybe golang would be a better language to drive the benchmarks instead of shell?  I would have used bash too, but maybe other folks prefer something more testable?

Reply:
Will work on migrating to Go after this PR